### PR TITLE
Alerting: updated alerting creation order

### LINKED ISSUE #48483: Move alert type picker, query and expression editor and alert condition to Step 1
Move alert type picker, query and expression editor and alert condition to Step 1

### PR DESCRIPTION
**Pull request by gillesdemey**
**What this PR does / why we need it**:

This PR updates the alert creation flow sequence by moving the alert rule details and namespaces / groups to the last step of the alert creation.

https://www.figma.com/file/2s7EGevGeI4GuiArOmUECK/Unified-Alerting?node-id=3852%3A24160

Fixes #48483 related to #48247

**Special notes for your reviewer**:

Some files have been renamed and moved around to keep the filesystem structure aligned with the creation flow – more than happy to change the file and folder names to other proposals :v:

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/ConditionField.tsx [hunk 1]
```diff
@@ -31,10 +32,13 @@ export const ConditionField: FC = () => {
...
-      setValue('condition', expressions.length ? expressions[expressions.length - 1].refId : null);
-    } else if (!condition && expressions.length) {
-      setValue('condition', expressions[expressions.length - 1].refId);
+    const lastExpression = last(expressions);
+    const conditionExists = options.find(({ value }) => value === condition);
+
+    if (condition && !conditionExists) {
+      setValue('condition', lastExpression?.refId ?? null);
+    } else if (!condition && lastExpression) {
+      setValue('condition', lastExpression.refId, { shouldValidate: true });
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rule-editor/ConditionField.tsx:41]: This fixes a subtle bug where you would not be able to preview anything if you haven't changed any queries or manually selected a condition from the dropdown

File: public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx [hunk 2]
```diff
@@ -151,11 +150,10 @@ export const AlertRuleForm: FC<Props> = ({ existing }) => {
         <div className={styles.contentOuter}>
           <CustomScrollbar autoHeightMin="100%" hideHorizontalTrack={true}>
             <div className={styles.contentInner}>
-              <AlertTypeStep editingExistingRule={!!existing} />
+              <QueryAndAlertConditionStep editingExistingRule={!!existing} />
               {showStep2 && (
```
Comment (peterholmberg) [inline public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx:154]: Given that we default to Grafana managed rule, is this check needed still?

**Review by peterholmberg — COMMENTED** [review]
This looks great, feels like a much nicer experience. Only one nit.

File: public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx (hunk 2 above)
Comment (peterholmberg) [reply public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx:154]: Ah right

**Review by peterholmberg — APPROVED** [review]
:shipit:

**Pull request merged**